### ISSUE TITLE
Queries return undefined when entities have expired

### ISSUE THREAD

**Issue by Amnesthesia** [NerdWalletOSS/apollo-cache-policies#70]
When we use the `timeToLive` setting, entities seem to be expiring in the cache — but, after expiring, even though the query is correctly firing a network call again the next time it is requested (although, would have preferred it to automatically fire again as soon as the cache expires), the `data` object received is `undefined`

This is super strange, because I can see the request being fired and a response being returned, but nothing is received by Apollo — it's as if the new response isn't being written to the cache? Perhaps that query is stuck with an old cache key?

**Comment by danReynolds** [comment]
Thanks for the report! So a query expires via  TTL, it goes to network to get the data again, but it never returns that data? Is there a repro I could look into?

**Comment by Amnesthesia** [comment]
@danReynolds Yeah that's exactly what's happening — I see the network request, and `const { data } = useQuery(..)` is undefined, even though the request just came through. Disabling TTL again fixes it

Unfortunately our repo is private but I tried to reproduce the behaviour here, and can get similar behaviour: 

https://codesandbox.io/s/suspicious-goldstine-rw2ggr?file=/src/graphql/query.tsx

<img width="1065" alt="Screenshot 2023-08-28 at 08 20 24" src="https://github.com/NerdWalletOSS/apollo-cache-policies/assets/3265008/d00b6156-38c4-4ed9-8eed-a7f273f73d0b">

**Comment by Amnesthesia** [comment]
@danReynolds any updates on this?
